### PR TITLE
[FIX] Relative/absolute record name conversions

### LINKED ISSUE #3: Relative/absolute record name conversions
Hi,

Over in the libdns repo, we're discussing standardizing the use of relative or absolute record names in `libdns.Record` structs. The current consensus is that they should be made relative (i.e. for a record named `sub.example.com` in a zone called `example.com`, the name should become `sub`). This may affect your provider implementation.

When you have a chance, would you please check out [this issue](https://github.com/libdns/libdns/issues/12#issuecomment-756135908)? This is the last call for feedback before we decide and I push the changes, and I want to make sure we get it right for the majority of DNS provider implementations so we don't have to change it again.

Link: https://github.com/libdns/libdns/issues/12#issuecomment-756135908

Thank you!

### LINKED ISSUE #4: This lib is broken when using caddy master
This issue Matt raises means this lib is broken:  https://github.com/libdns/route53/issues/3

So currently using the master branch of caddy the route53 plugin (which uses this library is all broken).

Route53 complains because the paths are not right.

Has anyone who knows  this code taken a look?

### PR DESCRIPTION
**Pull request by PhoenixPeca**
Hi @danlsgiga and @mholt , please help check my codes if I have introduced any logical errors.. I'm new to Golang.. spent my entire day working on this and to test. Feel free to edit or add recommendations.. I am aware of the incorrect module references (in go.mod and go.sum), will have them fixed after you confirm the codes. :D

Before this PR, this DNS module does not work due to recent changes in the libdns/libdns module.. I hope this will fix that issue and close #3 and close #4

It works perfectly on my side. :D
image

I have tested by building using this:
```
xcaddy build \
    --with github.com/caddy-dns/cloudflare \
    --with github.com/PhoenixPeca/caddy-dns-route53@v1.2.6
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,9 +1,8 @@
-module github.com/libdns/route53
+module github.com/PhoenixPeca/libdns-route53
```
Comment (francislavoie) [inline go.mod:1]: You'll need to revert this change before it can be merged

**Comment by PhoenixPeca** [comment]
Hi @francislavoie , I have resolved your inputs. :D Thank you for taking the time to review.

**Comment by PhoenixPeca** [comment]
Dear all, may I follow up on this one? :D

Thanks all!

**Comment by silversteez** [comment]
Hello! I'm currently getting the [same error](https://github.com/libdns/route53/issues/3#issuecomment-821978373) as @PhoenixPeca - I'm guessing the problem is the same and it sounds like this PR is the fix.

Would be very appreciative if this could be reviewed and/or merged! Thanks! cc @francislavoie

**Comment by blarghmatey** [comment]
Count me in as another user that would like to see this PR merged as I'm also running up against this bug.

**Comment by PhoenixPeca** [comment]
> Update - I've tried using @PhoenixPeca's fork (`--with github.com/PhoenixPeca/caddy-dns-route53`) and it fixed the problem. So it does appear that something is currently broken in `caddy-dns/route53` and this fixes it.
> 
> Thank you @PhoenixPeca for working on this, btw!

Hey @silversteez , cool! that's good to know. :D

**Review by mholt — APPROVED** [review]
Going to preempt the owner on this one, since owner is MIA and this seems like a good patch.

**Comment by mholt** [comment]
@PhoenixPeca Would you be interested in being a comaintainer of this repo? (It could use a new release.)

**Comment by PhoenixPeca** [comment]
> @PhoenixPeca Would you be interested in being a comaintainer of this repo? (It could use a new release.)

Hi @mholt , I think that would be great! :D Will do my best to help maintain.. ;)

**Comment by PhoenixPeca** [comment]
Hi @mholt , I think I may also need access with the caddy-dns/route53 repo since I will need to bump up the version there as well:

https://github.com/caddy-dns/route53/blob/cb329a5c3ab0a49b8e8fee1cccb3fa621ef1902b/go.mod#L7


✌️

**Comment by danlsgiga** [comment]
hey folks, sorry about the huge delay! Some changes in life here and it will become really challenging to keep maintaining this repo. Thanks for taking it onwards @PhoenixPeca and @mholt!!
